api/api-doc/storage_service.json

```diff
@@ -2852,6 +2852,46 @@
             }
          ]
       },
+      {
+         "path":"/storage_service/tablets/repair",
+         "operations":[
+            {
+               "nickname":"repair_tablet",
+               "method":"POST",
+               "summary":"Repair a tablet",
+               "type":"void",
+               "produces":[
+                  "application/json"
+               ],
+               "parameters":[
+                  {
+                     "name":"ks",
+                     "description":"Keyspace name to repair",
+                     "required":true,
+                     "allowMultiple":false,
+                     "type":"string",
+                     "paramType":"query"
+                  },
+                  {
+                     "name":"table",
+                     "description":"Table name to repair",
+                     "required":true,
+                     "allowMultiple":false,
+                     "type":"string",
+                     "paramType":"query"
+                  },
+                  {
+                     "name":"tokens",
```

Comment:
Btw, this "tokens" param cannot be used to properly replace `nodetool repair` "ranges" option. If you obtain tokens from tablet metadata and then call the API with those tokens, tablet split may sneak in between and fewer tablets will be repaired, not the whole range.

The API only guarantees that those specific tokens are repaired. Maybe more, but there is no guarantee how much more.

Reply:
The ranges option is legacy. We are not going to support arbitrary range selection for tablet. 

As I mentioned in the previous discussion,  the split or merge is not an important factor for repair all tablets anymore,  we can look at the repair_time field for the tablets, and repair those tablets with "old" repair_time and advance them.

Reply:
> The ranges option is legacy. We are not going to support arbitrary range selection for tablet.

Ok.

> As I mentioned in the previous discussion, the split or merge is not an important factor for repair all tablets anymore, we can look at the repair_time field for the tablets, and repair those tablets with "old" repair_time and advance them.

split/merge is not a problem for tokens=all or repairing a single token anyway. After task is submitted, split will preserve the task for split-off tablets.